src/cluster.c

```diff
@@ -4897,16 +5031,14 @@ int verifyClusterConfigWithData(void) {
          * assigned to this slot. Fix this condition. */
 
         update_config++;
-        /* Case A: slot is unassigned. Take responsibility for it. */
+        /* slot is unassigned. Take responsibility for it. */
         if (server.cluster->slots[j] == NULL) {
             serverLog(LL_WARNING, "I have keys for unassigned slot %d. "
                                     "Taking responsibility for it.",j);
             clusterAddSlot(myself,j);
-        } else {
+        } else if (server.cluster->importing_slots_from[j] != server.cluster->slots[j]) {
```

Comment:
Should we clear the data out if we aren't going to import it? Otherwise we just have random stale data.